app/screens/FollowInfoStorage.js

```diff
@@ -8,63 +8,124 @@ export async function setToken(token) {
+    locations_followed = getLocationsFollowed();  
     locations_followed.push(location_id);                                                        
+    updateBackend();
     return await AsyncStorage.setItem('locations_followed', JSON.stringify(locations_followed));
 }
 
 export async function unfollowLocation(location_id) {
-    locations_followed = JSON.parse(await AsyncStorage.getItem('locations_followed') || "[]"); 
+    locations_followed = getLocationsFollowed(); 
     var index = locations_followed.indexOf(location_id);
     if (index !== -1) locations_followed.splice(index);                                       
+    updateBackend();
     return await AsyncStorage.setItem('locations_followed', JSON.stringify(locations_followed));
 }
 
 export async function isFollowingCategory(category_id) {
-    categories_followed = JSON.parse(await AsyncStorage.getItem('categories_followed') || "[]");
+    categories_followed = getCategoriesFollowed()
     var index = categories_followed.indexOf(category_id);
     if (index !== -1) return true;
     else return false;
 }
 
 export async function isFollowingAuthor(author_id) {
-    authors_followed = JSON.parse(await AsyncStorage.getItem('authors_followed') || "[]");
+    authors_followed = getAuthorsFollowed();
     var index = authors_followed.indexOf(author_id);
     if (index !== -1) return true;
     else return false;
 }
 
 export async function isFollowingLocation(location_id) {
-    locations_followed = JSON.parse(await AsyncStorage.getItem('locations_followed') || "[]");
+    locations_followed = getLocationsFollowed();
     var index = locations_followed.indexOf(location_id);
     if (index !== -1) return true;
     else return false;
 }
 
+export async function addNotificationSetting(notification_id) {
+    notification_settings = getNotificationSettings();  
+    notification_settings.push(notification_id);                                                        
+    updateBackend();
+    return await AsyncStorage.setItem('notification_settings', JSON.stringify(notification_settings));
+}
+
+export async function removeNotificationSetting(notification_id) {
+    notification_settings = getNotificationSettings(); 
+    var index = notification_settings.indexOf(notification_id);
+    if (index !== -1) notification_settings.splice(index);                                     // remove notification option from list
+    updateBackend();
+    return await AsyncStorage.setItem('notification_settings', JSON.stringify(notification_settings));
+}
+
+export async function isBeingNotified(notification_id) {
+    notification_settings = getNotificationSettings();
+    var index = notification_settings.indexOf(notification_id);
+    if (index !== -1) return true;
+    else return false;
+}
+
+async function updateBackend() {
+    fetch('https://www.stanforddaily.com/wp-json/tsd/v1/push-notification/users/' + await getToken(), {
```

Comment:
You want this function to return a promise.
```suggestion
    return fetch('https://www.stanforddaily.com/wp-json/tsd/v1/push-notification/users/' + await getToken(), {
```